### ISSUE TITLE
Requests skipped and bloc events not firing after reopening app

### ISSUE THREAD

**Issue by TedRibeiro** [flutterchina/dio#947]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   Flutter   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |   1.20.4   |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |   3.0.10   |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |   VS Code   |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |   100%   |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |   N/A   |
| Demo project link              | e.g. link to a demo project that highlights the issue |   N/A   |

### Issue Description and Steps

All requests runs flawlesly on app first run. 
But if I close the app and then reopen it (or even if I rerun it through `flutter run`), the first request is always skipped. It even hits the point where it is called but I don't get any response at all. Actually, the request is never made (checked it using an inspector on my backend) and it acts like the function reached its end, not hitting any of lines after the request and not having a response nor exception. But the following requests work fine.

Another strange behavior that happens after reopening the app: after a given request is completed, if I add a new event to a bloc, it is never triggered.

None of these behaviors happen during the first run or using the default HTTP library.

**Comment by jrc2139** [comment]
I'm having the same issue, but it's only with certain Android devices.  The difference is that it won't work at first.  I have to close and reopen the app about 6 times before it will complete the call.

**Comment by TedRibeiro** [comment]
An update on my case: My problem wasn't related to Dio library at all. I was using an outdated version of Geolocator library. My http requests were awaiting for fetching the location, which never completed. 
@jrc2139 you should check on the libraries you've been using

**Comment by jrc2139** [comment]
@TedRibeiro thanks for the advice!  the issue ended up being device os specific.  there were other apps constantly crashing in its background that would cause my app to restart and lose its state upon returning from recents.  oh well!
